src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -73,7 +76,13 @@ public function getItemFromIri(string $iri, array $context = [])
             throw new InvalidArgumentException(sprintf('No resource associated to "%s".', $iri));
         }
 
-        if ($item = $this->itemDataProvider->getItem($parameters['_api_resource_class'], $parameters['id'], null, $context)) {
+        $identifiers = [];
+
+        if ($this->identifierNormalizer) {
```

Comment:
Can't we do this directly in the data provider? It would encapsulate the logic in only one place.